modules/python/Makefile.style

```diff
@@ -44,26 +46,74 @@ python/autoflake: bh/venv
 .PHONY: python/isort/check
 ## Check python imports using isort
 python/isort/check:
-	@$(DOCKER) pull alphachai/isort
 	@$(MAKE) python/lint/announce/isort
+	@if [[ -n "$(JQ)" ]] && [[ -n "$$(jq -r <Pipfile.lock '.develop | to_entries[] | .key + .value.version' | grep isort)" ]]; then \
+		$(MAKE) python/isort/check/pipenv; \
+	else \
+		$(MAKE) python/isort/check/docker; \
+	fi;
+
+.PHONY: python/isort/check/pipenv
+python/isort/check/pipenv:
+	$(PIPENV_RUN) isort $(ISORT_FLAGS) --check-only $(PYTHON_LINT_TARGETS) --diff
+
+.PHONY: python/isort/check/docker
+python/isort/check/docker:
+	@$(DOCKER) pull alphachai/isort
 	@$(DOCKER) run -i --rm --user $$(id -u):$$(id -g) --mount type=bind,src=$$(pwd),dst=/app alphachai/isort $(ISORT_FLAGS) --check-only $(PYTHON_LINT_TARGETS) --diff
 
 .PHONY: python/isort
 ## Fix python imports using isort
 python/isort:
+	@if [[ -n "$(JQ)" ]] && [[ -n "$$(jq -r <Pipfile.lock '.develop | to_entries[] | .key + .value.version' | grep isort)" ]]; then \
+		$(MAKE) python/isort/pipenv; \
+	else \
+		$(MAKE) python/isort/docker; \
+	fi;
+
+.PHONY: python/isort/pipenv
+python/isort/pipenv:
+	$(PIPENV_RUN) isort $(ISORT_FLAGS) $(PYTHON_LINT_TARGETS)
+
+.PHONY: python/isort/docker
+python/isort/docker:
 	@$(DOCKER) pull alphachai/isort
 	$(DOCKER) run -i --rm --user $$(id -u):$$(id -g) --mount type=bind,src=$$(pwd),dst=/app alphachai/isort $(ISORT_FLAGS) $(PYTHON_LINT_TARGETS)
 
 .PHONY: python/black/check
 ## Check python files using black
 python/black/check:
-	@$(DOCKER) pull alphachai/black
 	@$(MAKE) python/lint/announce/black
-	@$(DOCKER) run -i --rm --user $$(id -u):$$(id -g) --mount type=bind,src=$$(pwd),dst=/app alphachai/black --check $(PYTHON_LINT_TARGETS)
+	@if [[ -n "$(JQ)" ]] && [[ -n "$$(jq -r <Pipfile.lock '.develop | to_entries[] | .key + .value.version' | grep black)" ]]; then \
+		$(MAKE) python/black/check/pipenv; \
+	else \
+		$(MAKE) python/black/check/docker; \
+	fi;
+
+.PHONY: python/black/check/pipenv
+python/black/check/pipenv:
+	$(PIPENV_RUN) black --check $(PYTHON_LINT_TARGETS)
+
+.PHONY: python/black/check/docker
+python/black/check/docker:
+	@$(DOCKER) pull alphachai/black
+	$(DOCKER) run -i --rm --user $$(id -u):$$(id -g) --mount type=bind,src=$$(pwd),dst=/app alphachai/black --check $(PYTHON_LINT_TARGETS)
```

Comment:
We could use`cytopia/black:latest` which gets built regularly or I can setup `mintel/black`

Reply:
cytopia/black looks fine

Reply:
orrrr, we could upgrade alphachai/black to build every day :P